components/LogSummary.js

```diff
@@ -0,0 +1,121 @@
+import moment from 'moment'
+import { Component } from 'react'
+import { withAuth } from 'use-auth0-hooks'
+
+import { secureFetch } from '../util'
+
+class LogSummary extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      logs: null,
+      logsError: null
+    }
+    // TODO fix babel plugin so we can use class properties
+    // https://babeljs.io/docs/en/babel-plugin-proposal-class-properties
+    // https://nextjs.org/docs/advanced-features/customizing-babel-config
+    this.handleFetchLogs = this.handleFetchLogs.bind(this)
+  }
+
+  async handleFetchLogs (force = false) {
+    const { logs, logsError } = this.state
+    if (!force && (logs || logsError)) {
+      return
+    }
+
+    const { accessToken } = this.props.auth
+    if (!accessToken) {
+      return
+    }
+    const fetchedLogs = await secureFetch(`${process.env.API_BASE_URL}/api/secure/logs?startDate=blah`, accessToken)
+    if (fetchedLogs) {
+      this.setState({ logs: fetchedLogs })
+    }
+  }
+
+  async componentDidMount () {
+    await this.handleFetchLogs()
+  }
+
+  async componentDidUpdate () {
+    await this.handleFetchLogs()
+  }
+
+  render () {
+    const { auth } = this.props
+    const { logs, logsError } = this.state
+    if (!auth.isAuthenticated) return null
+    const hasLogs = logs && logs.length > 0
+    return (
+      <div>
+        <h2>Log Summary</h2>
+        <button onClick={this.handleFetchLogs}>Fetch logs</button>
+        <a
+          target='_blank'
+          rel='noopener noreferrer'
+          href='https://console.aws.amazon.com/apigateway/home?region=us-east-1#/usage-plans'
+        >
+          Open AWS console
+        </a>
+        {
+          hasLogs && (
+            <div>
+              {logsError && <pre>Error loading logs: {logsError}</pre>}
+              <div>All requests made over the last 30 days</div>
+              {logs.map((plan, planIndex) => {
+                const keyIds = Object.keys(plan.items)
```

Comment:
(There is potential for a `LogEntry` component.)

Reply:
I think I'll leave this alone for now, but will add a TODO.